File: SimpleAnalyticsDemo-ObjC/SimpleAnalyticsDemo-ObjC/ViewController.m
//
//  ViewController.m
//

#import "ViewController.h"
#import "AnalyticsManager.h"

@implementation ViewController

- (void)viewDidLoad {
    [super viewDidLoad];

    [AnalyticsManager initializeEndpoint];
    [AnalyticsManager addAnalyticsItem:@"Displayed main view" details:nil];
}

- (IBAction)handleSoundButton:(NSButton *)sender {
    [self playSound:sender.title];
}

- (IBAction)repeatLast:(NSButton *)sender {
    [AnalyticsManager addCounter:@"Count Something"];
}

- (IBAction)repeatLastOther:(NSButton *)sender {
    [AnalyticsManager addCounter:@"Count Something Else"];
}

- (void)playSound:(NSString *)title {
    NSLog(@"Played sound: %@", title);
    NSString * randString = [self randomString];
    NSDictionary *demoDetail = @{@"Demo detail":randString};
    [AnalyticsManager addAnalyticsItem:title details:demoDetail];
}

- (NSString *)randomString {
    NSArray* words = [@"The quick brown fox jumped over the cow" componentsSeparatedByString:@" "];
    NSUInteger count = words.count;
    int index = arc4random_uniform((UInt32)count);
    return  words[index];
}
@end

File: SimpleAnalyticsDemo-AppKit/SimpleAnalyticsDemo-AppKit/ViewController.swift
//
//  ViewController.swift
//  SimpleAnalyticsDemo-AppKit
//
//  Created by Dennis Birch on 4/14/21.
//

import Cocoa

class ViewController: NSViewController {
    override func viewDidLoad() {
        super.viewDidLoad()

        DemoAnalytics.addAnalyticsItem("Display main view")
    }

    @IBAction func handleButtonPress(_ sender: NSButton) {
        let name = sender.title
        
        if name.lowercased().contains("repeat") {
            DemoAnalytics.countItem(name)
        } else {
            DemoAnalytics.addAnalyticsItem(name, params: ["Demo detail" : randomString()])
        }
    }

    private func randomString() -> String {
        let wordArray = "The quick brown fox jumped over the cow".split(separator: " ")
        let count = wordArray.count
        let index = arc4random_uniform(UInt32(count))
        return String(wordArray[Int(index)])
    }
}

